coderd/notifications/dispatch/smtp.go

```diff
@@ -93,37 +104,24 @@ func (s *SMTPHandler) dispatch(subject, htmlBody, plainBody, to string) Delivery
 		default:
 		}
 
-		var (
-			c    *smtp.Client
-			conn net.Conn
-			err  error
-		)
-
 		s.log.Debug(ctx, "dispatching via SMTP", slog.F("msg_id", msgID))
 
 		// Dial the smarthost to establish a connection.
 		smarthost, smarthostPort, err := s.smarthost()
 		if err != nil {
 			return false, xerrors.Errorf("'smarthost' validation: %w", err)
 		}
-		if smarthostPort == "465" {
-			return false, xerrors.New("TLS is not currently supported")
-		}
 
-		var d net.Dialer
 		// Outer context has a deadline (see CODER_NOTIFICATIONS_DISPATCH_TIMEOUT).
-		conn, err = d.DialContext(ctx, "tcp", fmt.Sprintf("%s:%s", smarthost, smarthostPort))
-		if err != nil {
-			return true, xerrors.Errorf("establish connection to server: %w", err)
+		if _, ok := ctx.Deadline(); !ok {
+			return false, xerrors.Errorf("context has no deadline")
 		}
 
-		// Create an SMTP client.
-		c, err = smtp.NewClient(conn, smarthost)
+		// TODO: reuse client across dispatches (if possible).
```

Comment:
I'm afraid we have to implement it now (or in follow-up as this PR is large). SMTP servers may have enabled restrictions limiting login attempts.

Reply:
I doubt it; Alertmanager does it this way and I'm not aware of any issues arising from that. I don't think we'll need this for the initial version.

Reply:
I see, so it will just operate slower. Let's leave it then.

Reply:
Added https://github.com/coder/team-coconut/issues/15